File: rotate.js
// Given input
let matrix1 =
[
  [1,2,3],
  [4,5,6],
  [7,8,9]
];
//
// rotate the input matrix in-place such that it becomes:
// [
//   [7,4,1],
//   [8,5,2],
//   [9,6,3]
// ]

function rotate(matrix) {
  for (let row = 0; row < matrix.length; row++) {
    for (let col = 0; col < matrix.length; col++) {
      matrix[row].push(matrix[matrix.length - 1 - col][row]);
    }
  }
  for (let row of matrix) {
    row.splice(0,matrix.length);
  }
}

console.log(matrix1);
rotate(matrix1);
console.log(matrix1);

File: rotate.rb
matrix1 =
[
  [ 5, 1, 9,11],
  [ 2, 4, 8,10],
  [13, 3, 6, 7],
  [15,14,12,16]
]

def rotate(matrix)
  row = 0
  while row < matrix.length
    col = 0
    while col < matrix.length
      matrix[row] << matrix[matrix.length - 1 - col][row]
      col += 1
    end
    row += 1
  end

  matrix.each do |row|
    row.slice!(0,matrix.length)
  end
end

p matrix1
rotate(matrix1)
p matrix1
